openspec/changes/add-model-provider-cred-store-oagw/specs/cred-store/spec.md

```diff
@@ -0,0 +1,843 @@
+
+The plugin SDK SHALL define a plugin spec schema type (for plugin instance registration) as a derived GTS schema:
+- `CredStorePluginSpecV1` with schema id `gts.x.core.modkit.plugin.v1~x.core.cred_store.plugin.v1~`
+
+Each plugin module SHALL register exactly one or more plugin instances in `types_registry` using:
+- `BaseModkitPluginV1<CredStorePluginSpecV1>`
+
+Each registered plugin instance SHALL include:
+- `id: GtsInstanceId` (full instance id)
+- `vendor: string`
+- `priority: int` (lower wins)
+- `properties: CredStorePluginSpecV1` can be empty
+
+Each plugin module SHALL also register a scoped client in `ClientHub` under `ClientScope::gts_id(&instance_id)`.
+
+Only one plugin SHALL be active at runtime: the eligible plugin with the lowest `priority` value among registered instances.
+
+Eligibility MUST be determined by priority (lower priority wins).
+
+#### Scenario: Single active plugin
+- **GIVEN** multiple cred_store plugins are registered
+- **WHEN** the gateway resolves its plugin
+- **THEN** it chooses the plugin with the lowest priority
+
+#### Scenario: Scoped client resolution
+- **GIVEN** the gateway selected plugin instance id `gts.x.core.modkit.plugin.v1~x.core.cred_store.plugin.v1~vendor.pkg.plugin.v1`
+- **WHEN** the gateway requests a client via `ClientHub.get_scoped::<dyn CredStorePluginApi>(ClientScope::gts_id(&plugin_instance_id))`
+- **THEN** the plugin client is resolved and used to serve the request
+
+### Requirement: cred_store plugin persistence, access control, and cryptography
+The system SHALL define how `cred_store` plugins persist secrets, enforce access control, and apply cryptography.
+
+The gateway SHALL be responsible for:
+- Performing request authentication and high-level authorization (role/permission checks) before calling the plugin.
+- Passing request-scoped `SecurityCtx` into the plugin.
+
+Each plugin SHALL be responsible for:
+- Enforcing tenant/user scoping when it interacts with any persistent store.
+- Ensuring that secret material is never logged.
+
+#### Access control (DB-backed plugins)
+For DB-backed plugins, access control SHALL be enforced using the `modkit-db` secure ORM layer.
+
+DB-backed plugins SHALL:
+- Use a secure DB access wrapper (e.g., `SecureConn`) for all queries.
+- Derive an `AccessScope` from the request `SecurityCtx` and apply it to all queries.
+- Use scoped CRUD operations that automatically produce deny-by-default behavior for empty scopes.
+
+#### Scenario: DB-backed plugin denies cross-tenant reads
+- **GIVEN** a secret record exists for tenant A with secret id `S`
+- **WHEN** a caller from tenant B calls `CredStorePluginApi::get_secret_material(ctx, S, secret_type_id)`
+- **THEN** the plugin query is scoped to tenant B
+- **AND** the plugin returns a not-found or permission-denied error
+
+#### Cryptography (DB-backed plugins)
+DB-backed plugins SHALL encrypt secret material before persisting it.
```

Comment:
This implies plugins store the ciphertext as the source of truth. Right now you are designing two owners for the same data. Decide the ownership model:

Either gateway owns encryption and stores ciphertext, plugin is thin storage.

OR plugin owns encryption and stores ciphertext, gateway stores only metadata and plugin references.
Do not do both unless you want double-encryption and split-brain.